src/middlewared/middlewared/plugins/pool.py

```diff
@@ -432,33 +431,7 @@ async def is_upgraded(self, oid):
 
     @private
     async def is_upgraded_by_name(self, name):
-        proc = await Popen(
-            f'zpool get -H -o value version {name}',
-            stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True
-        )
-        res, err = await proc.communicate()
-        if proc.returncode != 0:
-            return True
-        res = res.decode('utf8').rstrip('\n')
-        try:
-            int(res)
-        except ValueError:
-
-            if res == '-':
-                proc = await Popen(
-                    f"zpool get -H -o property,value all {name}",
-                    stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True
-                )
-                data = (await proc.communicate())[0].decode('utf8').strip('\n')
-                for line in [i for i in data.split('\n') if i.startswith('feature') and '\t' in i]:
-                    prop, value = line.split('\t', 1)
-                    if value not in ('active', 'enabled'):
-                        return False
-                return True
-            else:
-                return False
-        else:
-            return False
+        return await self.middleware.call('zfs.pool.is_upgraded', name)
```

Comment:
This method should either keep returning `False` for unavailable pools or all its invocations should handle `CallError` (e.g. here https://github.com/truenas/middleware/blob/a26e1bc815b21cf85f5c8cfa71611ae809a5ead1/src/middlewared/middlewared/plugins/pool.py#L558 it might be called for a non-existing pool that is only present in the configuration database).